CONTRIBUTING.md

```diff
@@ -1,43 +1,122 @@
+
+### Writing Good Bug Reports and Feature Requests
+When reporting [issues](https://github.com/osgi/osgi.enroute.site/issues) on GitHub please file a single issue per problem and feature request. Do not file combo issues. Please do not submit multiple comments on a single issue - write your issue with all the environmental information and reproduction steps so that an engineer can reproduce it.
+
+An bug report could look similar to this:
+
+```
+Reproduction Steps:
+
+ 1. Do Something
+ 2. Do something else aaand...
+ 3. ... break things
+
+Observed behavior:
+	Console is exploding
+
+Expected behavior:
+	Console should NOT explode!
+
+Environment/Tools:
+	OS: Windows 10
+	IDE: Eclipse Oxygen
+	Java: 1.8.0
+	Gradle: 4.1
+	...
+
+Additional information:
+	anything else that might be useful (e.g. Screenshots)
+```
 This information will help us review and fix your issue faster.
+Feature requests usually can be less formal ;)
 
-## Building
+## Contributing Improvements
 
-The site uses [GitHub Pages](https://help.github.com/articles/what-are-github-pages/) with
-the site content in the `gh-pages` branch. To build the site locally, you will need to
-[install Ruby, Bundler and Jekyll](https://help.github.com/articles/using-jekyll-with-pages/#installing-jekyll).
+If you are not satisfied with only reporting issues and want to get your hands dirty, then rest assured: Any help is appreciated! We are always thrilled to receive pull requests, and do our best to process them as fast as possible. Not sure if that typo is worth a pull request?
+**Do it!**
+**We will appreciate it.**
 
-Run jekyll in the root directory to build:
+The site uses [GitHub Pages](https://help.github.com/articles/what-are-github-pages/) with the site content in the `gh-pages` branch. To build the site locally, you will need to install Ruby, Bundler and Jekyll. If you've never worked with GitHub Pages before, just keep reading.
 
-	$ bundle exec jekyll build
+Or you already are a pro at working with GitHub Pages? You have set up the [git triangular workflow](https://www.sociomantic.com/blog/2014/05/git-triangular-workflow/) for this repository and have Ruby, Bundler and Jekyll running? Then what are you waiting for? Read the [contributing guidelines](#contributing-guidelines) section and start writing!
 
-You can also run a local server to test the site:
+Github pages can be created either with usual HTML or you can use [Jekyll](https://jekyllrb.com/), a static web site generator, which takes [Markdown](https://jekyllrb.com/docs/posts/) files and transforms them into a static webpage. Usually you want to write in markdown, as this is way more convenient than writing pure HTML.
```

Comment:
I don't think the link for Markdown is good link to explain what markdown is. https://daringfireball.net/projects/markdown/ is a better one.